src/components/current_weather_display.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import Cloudy from '../images/cloudy.png'
+import Sunny from '../images/sunny.png'
+import Rainy from '../images/rainy.png'
+import Night from '../images/moon.png'
+import NightCloudy from '../images/nightclouds.png'
+import NightRain from '../images/night-rain.png'
+
+class CurrentWeatherDisplay extends React.Component {
+
+toFahrenheit(temp) {
+    return ((temp - 273.15) * 9/5 + 32).toFixed(0) + `\u00B0F`
+}
+
+toCelsius(temp) {
+    return (temp - 273.15).toFixed(0) + `\u00B0C`
+}
+
+render() {
+
+    const { cityName, currentDesc, currentWeather } = this.props 
+  
+    const currTime = (new Date().toLocaleString()).substring(10, 24)
+    let currHour = parseInt(currTime.substring(0,2))
+    let PM = currTime.substring(9,11)
+
+    const display = currentWeather && currentDesc ? (
+            <>
+          <div className="city-name">{cityName}</div>
+            <div className="current-weather">
+                <div className="current-temp">{this.props.Fahrenheit ? this.toFahrenheit(currentWeather.temp) : this.toCelsius(currentWeather.temp)}</div>
+                {(((currHour >= 7 && PM === "PM") || (currHour < 5 && PM === "AM") || (currHour === 12 && PM === "AM"))) ? (
+                        <img className="forecast-img" alt="" src={currentDesc["0"].main === 'Clear' ? Night : currentDesc["0"].main === 'Rain' ? NightRain : NightCloudy}></img> ) : (
+                        <img className="forecast-img" alt="" src={currentDesc["0"].main === 'Clear' ? Sunny : currentDesc["0"].main === 'Rain' ? Rainy : Cloudy}></img> 
+                        )
+                }
```

Comment:
Your app is very clean and thorough, which is a huge plus! But, there are a few instances in the render function that are a bit hard to read. How would you refactor this with readability in mind?

Reply:
Thanks! I definitely see what you're talking about; to make these easier to read and understand, I could create smaller functions to evaluate, and reduce the complexity within the render function like this:

<img width="551" alt="Screen Shot 2020-11-11 at 10 03 18 PM" src="https://user-images.githubusercontent.com/19714255/98890535-39377080-246a-11eb-86b4-e1556c5d187d.png">
<img width="508" alt="Screen Shot 2020-11-11 at 10 03 34 PM" src="https://user-images.githubusercontent.com/19714255/98890542-3ccaf780-246a-11eb-9d53-f5d4f77758d4.png">
<img width="886" alt="Screen Shot 2020-11-11 at 10 03 57 PM" src="https://user-images.githubusercontent.com/19714255/98890550-40f71500-246a-11eb-85bc-537461630b98.png">
<img width="906" alt="Screen Shot 2020-11-11 at 10 04 07 PM" src="https://user-images.githubusercontent.com/19714255/98890559-45233280-246a-11eb-82da-d589ecd14b77.png">

Reply:
Nice job! I'd recommend looking into some more functional patterns rather than conditional logic in this situation but it's much more readable this way.